docker-app/qfieldcloud/core/tests/test_api.py

```diff
@@ -71,9 +72,24 @@ def test_api_pagination_limitoffset(self):
         self.assertEqual(len(results_with_offset), page_size)
 
         # Obtain without pagination (= control test)
-        results_without_pagination = self.client.get(
+        results_without_offset_or_request_level_limit = self.client.get(
             "/api/v1/projects/",
         ).json()
 
-        # Test length (this is super slow -- 1 minute or so -- because of serialization)
-        self.assertEqual(len(results_without_pagination), unlimited_count)
+        # Even though the request is not setting a limit, the Project modelviewset
+        # was defined with a default limit that's kicking in here
+        self.assertEqual(
+            ProjectViewSet.pagination_class.default_limit,
+            len(results_without_offset_or_request_level_limit),
+        )
+
+    def test_api_headers_count(self):
+        """Test LimitOffset pagination custom 'X-Total-Count' headers implementation"""
+        # Authenticate client
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.token.key)
+
+        response = self.client.get("/api/v1/projects/")
+        self.assertEqual(
+            int(response.headers["X-Total-Count"]),
+            self.total_projects,
```

Comment:
Fixed: count from queryset, not from paginated results

Reply:
Thanks!